pkg/ccl/changefeedccl/show_changefeed_jobs_test.go

```diff
@@ -116,6 +119,46 @@ func TestShowChangefeedJobsBasic(t *testing.T) {
 	cdcTest(t, testFn, feedTestOmitSinks("webhook", "sinkless"), feedTestNoExternalConnection)
 }
 
+// TestShowChangefeedJobsShowsHighWaterTimestamp verifies that SHOW CHANGEFEED
+// JOBS includes a readable_high_water_timestamp which is a readable timestamp
+// but otherwise corresponds to the HLC time in high_water_timestamp.
+func TestShowChangefeedJobsShowsHighWaterTimestamp(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	testFn := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+		sqlDB := sqlutils.MakeSQLRunner(s.DB)
+		sqlDB.Exec(t, `CREATE TABLE foo(a INT PRIMARY KEY)`)
+		foo := feed(t, f, `CREATE CHANGEFEED FOR foo WITH resolved='0s',min_checkpoint_frequency='0s'`)
+
+		defer closeFeed(t, foo)
+
+		var highWaterHLC gosql.NullFloat64
+		var readableHighWater gosql.NullTime
+
+		// Wait for the high water timestamp to be non-null.
+		testutils.SucceedsSoon(t, func() error {
+			stmt := `SELECT high_water_timestamp, readable_high_water_timestamptz from [SHOW CHANGEFEED JOBS]`
+			sqlDB.QueryRow(t, stmt).Scan(&highWaterHLC, &readableHighWater)
+
+			if !highWaterHLC.Valid {
+				return errors.Errorf("high water timestamp not populated: %v", highWaterHLC)
+			}
+
+			return nil
+		})
+
+		highWaterTimestamp := time.Unix(0, int64(highWaterHLC.Float64)).UTC()
+		// Timestamps in CockroachDB have microsecond precision by default.
+		roundedHighWaterTimestamp := highWaterTimestamp.Round(time.Microsecond)
+
+		differenceDuration := roundedHighWaterTimestamp.Sub(readableHighWater.Time).Abs()
```

Comment:
This approach looks fine to me. I was also thinking perhaps we can just do another `hlc_to_timestamp` and assert the results are the same as readableHighWater. Your call on whichever way you prefer. 

```
var hlcTimestamp time.Time
sqlConn.QueryRow(t, `SELECT hlc_to_timestamp(highWaterHLC)`).Scan(&hlcTimestamp)
```